events-api/src/modules/tickets/tickets.repository.ts

```diff
@@ -0,0 +1,52 @@
+import { TicketsCreateDto } from "./dto/requests/tickets-create-dto";
+import { db } from "#/database/database";
+import { v4 } from "uuid";
+import { TicketsModel } from "./tickets.model";
+
+export class TicketsRepository {
+  static async create(newTicket: TicketsCreateDto): Promise<void> {
+    const { eventId, ticketPrice, ticketQuantity } = newTicket;
+
+    await db.execute<TicketsModel>(
+      `
+        INSERT INTO tickets (ticket_id, ticket_quantity, ticket_price, event_id) 
+        VALUES (?, ?, ?, ?)
+      `,
+      [v4(), ticketQuantity, ticketPrice, eventId]
+    );
+  }
+
+  static async getAllByTicket(id: string): Promise<TicketsModel[]> {
+    const result = await db.execute<TicketsModel>(
+      `
+        SELECT
+            ticket_id as ticketId,
+            ticket_quantity as ticketQuantity,
+            ticket_price as ticketPrice
+        FROM
+            tickets
+        WHERE event_id = ?
+    `,
+      [id]
+    );
+
+    return result;
+  }
+
+  static async getTicketById(id: string): Promise<TicketsModel[]> {
+    const result = await db.execute<TicketsModel>(
+      `
+        SELECT
+            ticket_id as ticketId,
+            ticket_quantity as ticketQuantity,
+            ticket_price as ticketPrice
+        FROM
+            tickets
+        WHERE ticket_id = ?
+    `,
+      [id]
+    );
+
+    return result;
```

Comment:
It returns a list of records, you need to take only 1 record instead of returning a list of records

Reply:
but I am returing data as [] array no matter how many records

Reply:
But you are interested only in 1 record. Method is called getTicketById. Id is a primary key, so you either find 1 record or nothing. Array is incorrect return here. It should return an object or null

Reply:
done 
![Uploading image.png…]()